logging/logger_registry.go

```diff
@@ -173,3 +183,23 @@ func UpdateLoggerLevelWithCfg(name string) error {
 func GetCurrentConfig() []LoggerPatternConfig {
 	return globalLoggerRegistry.getCurrentConfig()
 }
+
+// GetOrNewLogger returns a logger with the specified name if it exists, otherwise it
+// creates and registers one with the same name.
+// Consider removing this as a part of or after RSDK-8291 (https://viam.atlassian.net/browse/RSDK-8291)
+// which may consolidate various calls to the rdk.networking logger into one.
+func GetOrNewLogger(name string) (logger Logger) {
+	logger, ok := globalLoggerRegistry.loggerNamed(name)
+	globalLoggerRegistry.mu.Lock()
+	defer globalLoggerRegistry.mu.Unlock()
+	if !ok {
```

Comment:
just noting that a logger can be created after we call `loggerNamed` but before we grab the write lock to create one ourselves - not sure how big of an issue that is.

Reply:
```suggestion
	globalLoggerRegistry.mu.Lock()
	defer globalLoggerRegistry.mu.Unlock()
	logger, ok := globalLoggerRegistry.loggers[name]
	if !ok {
```

Reply:
done